lib/yabeda/prometheus/exporter.rb

```diff
@@ -18,26 +18,51 @@ def call(env)
           @app.call(env)
         end
 
-        def start_metrics_server!(**rack_app_options)
+        def start_metrics_server!(start_in_thread: true, **rack_app_options)
+          if start_in_thread
+            start_server_in_thread!(**rack_app_options)
+          else
+            start_server_in_process!(**rack_app_options)
+          end
+        end
+
+        def start_server_in_thread!(**rack_app_options)
           Thread.new do
-            default_port = ENV.fetch("PORT", 9394)
-            ::Rack::Handler::WEBrick.run(
-              rack_app(**rack_app_options),
-              Host: ENV["PROMETHEUS_EXPORTER_BIND"] || "0.0.0.0",
-              Port: ENV.fetch("PROMETHEUS_EXPORTER_PORT", default_port),
-              AccessLog: [],
-            )
+            start_app(**rack_app_options)
+          end
+        end
+
+        def start_server_in_process!(**rack_app_options)
+          pid = Process.fork do
+            # re-configure yabeda since we're in a new process
+            Yabeda.configure!
```

Comment:
Will it really work?

I believe that if yabeda was already configured before fork, it won't allow to call this method.

See https://github.com/yabeda-rb/yabeda/blob/be2700fb3a3694b4f17c92d6c038ca37221d9ce7/lib/yabeda.rb#L91 (I believe instance variable values are preserved after fork)

Reply:
Yeah that’s what I thought too but i was seeing the same behaviour on prefork puma and passenger where if I don’t call configure! again in the process I get an empty page and no metrics are served via the endpoint from what I can tell it won’t serve metrics if they aren’t registered  on the specific process. Also if you want, you could pull it down and try it out. Perhaps where I’m installing our metrics is the problem? It’s in an initializer though which seems pretty typical a possible reason could also be because we’re using direct file store? I’ll try out one of the other stores and see if they have the same issue if not I could conditionally call configure!

Reply:
Figured out why sub-processes don't have the configured metrics declared, its because the `configure!`method is called in an `after_initialize` [hook](https://github.com/yabeda-rb/yabeda/blob/master/lib/yabeda/railtie.rb#L7) which never gets called in the sub process I'll make sure to check if its already configured like its being done here. If you're forking a process its pretty likely that you're not planning on running your entire app initialization process